tests/e2e/lib/dpa_helpers.go

```diff
@@ -155,13 +150,16 @@ func (v *DpaCustomResource) CreateOrUpdate(c client.Client, spec *oadpv1alpha1.D
 				},
 				Spec: *spec.DeepCopy(),
 			}
-			return v.Create(dpa)
```

Comment:
All changes from this line on, I think are unnecessary

the workflow is: UnsupportedOverrides will be in json, `tests/e2e/e2e_suite_test.go` will load the values, line 96 of this file will add values to DPAs created by test run

changes being done on this line on are duplication of was already done in line 96

Reply:
You are incorrect.  There is duplication in how the tests create the dpa.  My addition requires the above.  I was also confused about that.   DPA creation is handled in two different ways.
Follow the dpa creation from the two points:
* make test-e2e GINKGO_ARGS="--ginkgo.focus='MySQL application DATAMOVER'"
* make test-e2e GINKGO_ARGS="--ginkgo.focus='DPA reconciled to true'"

Should we perhaps fix that yes, but it's certainly NOT in scope for this change :)  1 scope per PR.

Reply:
Understood what you want to do, but I do not think this is the correct approach

Since tests here https://github.com/openshift/oadp-operator/blob/d5adc852931f1e23a24c7f0be7f86a27d107ce2d/tests/e2e/dpa_deployment_suite_test.go do not use `DpaCustomResource.Build` function, you were not seeing `UnsupportedOverrides` there

But this change breaks this test
https://github.com/openshift/oadp-operator/blob/d5adc852931f1e23a24c7f0be7f86a27d107ce2d/tests/e2e/dpa_deployment_suite_test.go#L362-L369

If I export `AWS_PLUGIN_IMAGE`, or not, the value defined in json will overwrite value defined in test

Reply:
The way I would go with this:
- remove the test mentioned above (since the idea is to run these tests with values of `UnsupportedOverrides` in the near future, so test would not be necessary anymore)
- remove this
  https://github.com/openshift/oadp-operator/blob/d5adc852931f1e23a24c7f0be7f86a27d107ce2d/tests/e2e/dpa_deployment_suite_test.go#L35
- change `testSpec.UnsupportedOverrides` here to `dpaCR.UnsupportedOverrides`
  https://github.com/openshift/oadp-operator/blob/d5adc852931f1e23a24c7f0be7f86a27d107ce2d/tests/e2e/dpa_deployment_suite_test.go#L69

Reply:
ok.. thank you for the in depth review!  I think I have made the changes you requested, I'm going to push an update and take some time to double check the changes and test.

Reply:
with the current changes, I think you can safely undo changes in lines 156-158 and line 165

the same changes are being applied to line 68 of `tests/e2e/dpa_deployment_suite_test.go`

Reply:
yes, I understand, however afaict you are still incorrect.
Here is what I see, and what happens when I test.
1.  line 68 of tests/e2e/dpa_deployment_suite_test.go will configure the dpa w/ overrides ONLY for the deployment test suite
2.  line 157/8 in dpa_helpers adds the overrides if a DPA exists at the time the backup_restore test suite runs
3.  line 165 in dpa_helpers adds the overrides if a DPA does NOT exist at the time the backup_restore test suite runs

Either follow the code paths, or test.. 
make test-e2e GINKGO_ARGS="--ginkgo.focus='DPA reconciled to true'"
w/ DPA make test-e2e GINKGO_ARGS="--ginkgo.focus='MySQL application CSI'"
w/o DPA make test-e2e GINKGO_ARGS="--ginkgo.focus='MySQL application CSI'"

Reply:
Reviewed the code again and the workflow that @mateusoliveira43 is raising it's because not all of the DPA CR's are generated using `Build()` function that set's the `UnsupportedOverrides`. The function itself is in:
   https://github.com/openshift/oadp-operator/blob/a5b0c0f9ed1c0cb41d745ab84e869635e87b5b07/tests/e2e/lib/dpa_helpers.go#L60-L97


We have two set's of tests which are taking advantage of the above function:
1. https://github.com/openshift/oadp-operator/blob/a5b0c0f9ed1c0cb41d745ab84e869635e87b5b07/tests/e2e/backup_restore_suite_test.go#L57
2. https://github.com/openshift/oadp-operator/blob/a5b0c0f9ed1c0cb41d745ab84e869635e87b5b07/tests/e2e/dpa_deployment_suite_test.go#L384

There is also `CreateOrUpdate` function where the DPA CR is being created outside of `Build` function, which also needs to be adjusted to take advantage of `UnsupportedOverrides`.
https://github.com/openshift/oadp-operator/blob/a5b0c0f9ed1c0cb41d745ab84e869635e87b5b07/tests/e2e/lib/dpa_helpers.go#L156-L164

So I do understand that the workflow may be simplified to have only one place to update `UnsupportedOverrides`, but I think it's outside of this PR and it's ok if it's being set even multiple times in favor of making smaller changes to the overall test functions.

We can have work to simplify this and ensure DPA CR is update via one generic function for all methods such as `Build` or `CreateOrUpdate`, but please correct me if I am wrong that it's outside of what it's being addressed by this PR?

Reply:
Yes, I wrote my comment to double that :)